src/cc/BPF.h

```diff
@@ -29,11 +29,6 @@
 
 namespace ebpf {
 
-enum class bpf_attach_type {
```

Comment:
could you keep it as 'enum bpf_attach_type' inside libbpf.h instead of introducing #define -s ?
compiler unfortunately consider int == enum for argument checking purpose, but it's still cleaner api.